src/coreclr/pal/src/arch/amd64/context2.S

```diff
@@ -183,6 +183,46 @@ LOCAL_LABEL(Done_Restore_CONTEXT_FLOATING_POINT):
     kmovq k6, qword ptr [rdi + (CONTEXT_KMask0 + 6 * 8)]
     kmovq k7, qword ptr [rdi + (CONTEXT_KMask0 + 7 * 8)]
 
+    test    BYTE PTR [rdi + CONTEXT_XStateFeaturesMask], XSTATE_MASK_APX
+    je      LOCAL_LABEL(Done_Restore_CONTEXT_XSTATE)
+
+    // TODO-XArch-APX:
+    // we are using raw hex code here to emit EGPRs-related changes,
+    // we will need to come back and re-write this part when assembler supports EGPRs.
```

Comment:
Is there some way to do an `#if` to only use the raw hex on assemblers without support today?

Reply:
I'm not aware of, do we have sth like LLVM version identifier defined in our code base?

I think this chunk is meant to be converted to actual `MOV` instructions once we have LLVM with official APX supports. 

@tannergooding, could you share some thoughts on this?

Reply:
I think its fine as is, I don't think its worth supporting code that we can't reliably test and maintain in CI at this point in time.

Rather, lets just have the one path and update when the toolchain can finally be updated to LLVM 19/20 or later

Reply:
I agree with this too.